examples/js/loaders/XfileLoader.js

```diff
@@ -0,0 +1,1518 @@
+
+                this.hierarchy.push(hierarchy_tmp[_i]);
+                var parentId = - 1;
+                for (var _m = 0; _m < this.hierarchy.length; _m++) {
+
+                    if (_i != _m && this.hierarchy[_i].parent === this.hierarchy[_m].name) {
+
+                        parentId = _m;
+                        break;
+
+                    }
+
+                }
+                this.hierarchy[_i].parent = parentId;
+
+            }
+
+        }
+    }, {
+        key: 'makeBonekeys',
+        value: function makeBonekeys(XAnimationInfo, bone, parent) {
+
+            var refObj = {};
+            refObj.name = bone;
+            refObj.parent = parent;
+            refObj.keys = [];
+            for (var i = 0; i < XAnimationInfo.keyFrames.length; i++) {
+
+                var keyframe = {};
+                keyframe.time = XAnimationInfo.keyFrames[i].time * this.fps;
+                keyframe.matrix = XAnimationInfo.keyFrames[i].matrix;
+                keyframe.pos = new THREE.Vector3().setFromMatrixPosition(keyframe.matrix);
+                keyframe.rot = new THREE.Quaternion().setFromRotationMatrix(keyframe.matrix);
+                keyframe.scl = new THREE.Vector3().setFromMatrixScale(keyframe.matrix);
+                refObj.keys.push(keyframe);
+
+            }
+            return refObj;
+
+        }
+    }]);
+    return XAnimationObj;
+
+}();
+
+var Xdata = function Xdata() {
+
+    classCallCheck(this, Xdata);
+
+    this.FrameInfo = [];
+    this.FrameInfo_Raw = [];
+    this.AnimationSetInfo = [];
+    this.AnimTicksPerSecond = 60;
+    this.XAnimationObj = null;
+
+};
+
+var XboneInf = function XboneInf() {
+
+    classCallCheck(this, XboneInf);
+
+    this.boneName = "";
+    this.BoneIndex = 0;
+    this.Indeces = [];
+    this.Weights = [];
+    this.initMatrix = null;
+    this.OffsetMatrix = null;
+
+};
+
+var XAnimationInfo$1 = XAnimationInfo = function XAnimationInfo() {
+
+    this.animeName = "";
+    this.boneName = "";
+    this.targetBone = null;
+    this.frameStartLv = 0;
+    this.keyFrames = [];
+    this.InverseMx = null;
+
+};
+
+var XFrameInfo$1 = XFrameInfo = function XFrameInfo() {
+
+    this.Mesh = null;
+    this.Geometry = null;
+    this.FrameName = "";
+    this.ParentName = "";
+    this.frameStartLv = 0;
+    this.FrameTransformMatrix = null;
+    this.children = [];
+    this.BoneInfs = [];
+    this.VertexSetedBoneCount = [];
+    this.Materials = [];
+
+};
+
+var XKeyFrameInfo = function XKeyFrameInfo() {
+
+    classCallCheck(this, XKeyFrameInfo);
+
+    this.index = 0;
+    this.Frame = 0;
+    this.time = 0.0;
+    this.matrix = null;
+
+};
+
+THREE.XFileLoader = function () {
+
+    function XFileLoader(manager, Texloader, _zflg) {
+
+        classCallCheck(this, XFileLoader);
+
+        this.manager = manager !== undefined ? manager : THREE.DefaultLoadingManager;
+        this.Texloader = Texloader !== undefined ? Texloader : THREE.TextureLoader;
```

Comment:
I mean, I'm wondering if you've missed 'new'

    this.Texloader = Texloader !== undefined ? Texloader : new THREE.TextureLoader(this.manager);